aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -520,7 +558,13 @@ void max_pool2d_with_indices_backward_out_cuda_template(
             block_x = std::min<int>(
                 maxThreadsDim[0], std::min<int>(lastPow2(nInputPlane), max_threads / block_y / block_z));
             const dim3 block(block_x, block_y, block_z);
-            int grid_x = nbatch;
+
+            int kernel_stride_C = cuda::ATenCeilDiv(
```

Comment:
I think it make sense to launch kernel on input. -> avoid atomic_add which is required when we launch kernel on output. Given that nhwc kernel is going to be targeting fp16, where atomic_add doesn't provide much benefit.

I would like to put a comment here, list our rational and the fact that we did not benchmark the atomic_add implementation (in case someone else re-visit the kernel).

Reply:
Added.